### PR TITLE
Create mocks for PlatformHookLib and PciLib

### PR DESCRIPTION
**Pull request by TsunFeng**
## Description

Create mocks for PlatformHookLib and PciLib

- [ ] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [x] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

Unit tests component can call PlatformHookLib and PciLib mock functions success

## Integration Instructions

N/A

### CODE REVIEW

**Review by os-d — APPROVED** [review]

File: MdeModulePkg/Test/Mock/Include/GoogleTest/Library/MockPlatformHookLib.h
Comment (makubacki) [inline MdeModulePkg/Test/Mock/Include/GoogleTest/Library/MockPlatformHookLib.h:1]: @TsunFeng, are the Intel copyrights appropriate for these new files?

**Review by makubacki — APPROVED** [review]

File: MdeModulePkg/Test/Mock/Include/GoogleTest/Library/MockPlatformHookLib.h
Comment (VivianNK) [reply MdeModulePkg/Test/Mock/Include/GoogleTest/Library/MockPlatformHookLib.h:1]: The library the file is mocking has intel copyright so I believe that transfers to the mock file, but we should add our copyright to the file like done here:
https://github.com/microsoft/mu_basecore/blob/45e4430db9d4a27a25201f51f91705fd7c4a0374/MdePkg/Test/Mock/Include/GoogleTest/Library/MockUefiDevicePathLib.h#L5

Unless you propose a different solution @makubacki ?

File: MdePkg/Test/MdePkgHostTest.dsc [hunk 1]
```diff
@@ -57,3 +57,4 @@
   MdePkg/Test/Mock/Library/GoogleTest/MockMemoryAllocationLib/MockMemoryAllocationLib.inf
   MdePkg/Test/Mock/Library/GoogleTest/MockPciExpressLib/MockPciExpressLib.inf
   MdePkg/Test/Mock/Library/GoogleTest/MockUefiDevicePathLib/MockUefiDevicePathLib.inf
+  MdePkg/Test/Mock/Library/GoogleTest/MockPciLib/MockPciLib.inf
```
Comment (VivianNK) [inline MdePkg/Test/MdePkgHostTest.dsc:60]: nit: looks like there's extra whitespace before the text

File: MdeModulePkg/Test/Mock/Include/GoogleTest/Library/MockPlatformHookLib.h
Comment (TsunFeng) [reply MdeModulePkg/Test/Mock/Include/GoogleTest/Library/MockPlatformHookLib.h:1]: Hi @makubacki,
Can I follow @VivianNK suggestion adding our copyright to the file?

thanks,
TF

**Review by VivianNK — APPROVED** [review]

File: MdeModulePkg/Test/Mock/Include/GoogleTest/Library/MockPlatformHookLib.h
Comment (makubacki) [reply MdeModulePkg/Test/Mock/Include/GoogleTest/Library/MockPlatformHookLib.h:1]: If you have authored the changes, please include your company's copyright. If you have taken the code with minimal modification, retain the original copyright.

File: MdePkg/Test/MdePkgHostTest.dsc (hunk 1 above)
Comment (TsunFeng) [reply MdePkg/Test/MdePkgHostTest.dsc:60]: I checked the format, and it looks the same as the previous line.

File: MdeModulePkg/Test/Mock/Include/GoogleTest/Library/MockPlatformHookLib.h
Comment (TsunFeng) [reply MdeModulePkg/Test/Mock/Include/GoogleTest/Library/MockPlatformHookLib.h:1]: Add Copyright (c) Microsoft Corporation. in the file

**Comment by TsunFeng** [comment]
Hi @VivianNK, @apop5 and @os-d
Could you help merge the PR to the base branch?

Thanks,
TF

**Pull request merged**